app/features-json/login_json_controller.rb

```diff
@@ -0,0 +1,48 @@
+require_relative "../shared/controller_base"
+require_relative "../services/user_service"
+require_relative "../services/dot_keys_variable_service"
+
+require "jwt"
+require "json"
+
+module FastlaneCI
+  # Controller responsible of handling the login process using JWT token.
+  class LoginJSONController < ControllerBase
+    HOME = "/login"
+
+    post HOME.to_s do
+      # Allow this endpoint to be requested with { Content-Type: application/json }
+      payload = params
+      payload = JSON.parse(request.body.read).symbolize_keys unless params[:path]
+      email = payload[:email]
+      password = payload[:password]
+      user = Services.user_service.login(email: email, password: password)
+      if user.nil?
+        halt(401)
+      else
+        content_type(:json)
+        { token: token(user) }.to_json
```

Comment:
can we add a TODO to also send the expiry of the token alongside the token.

Reply:
JWT has an expiry field [[1]](https://tools.ietf.org/html/rfc7519#section-4.1.4) that is already being sent (when you decode the token of course!) (now it lasts for one hour, again, can be changed 😄 )

Reply:
Does that mean the user needs to login after 1 hour or just a token refresh needed? If they need to login, we shouldn't expire their login but maybe every 6 months. We should rotate the auth token though, maybe daily, or weekly.

Reply:
JWT does not provide a refresh authentication token, if we want to support that, we should use OAuth on top of JWT in order to provide the `refresh_token` to the client. But as a matter of now, I can just make the expiration time like you said, 6 months long. Another thing, the key used to encrypt the token should be known for both the server and the client, (using the environment variable) how we manage that key rotation if not only the server knows how's the key rotating?

Reply:
@minuscorp Could you explain more on why the client needs to be able to decrypt the token?

Reply:
To be able to decode the payload of the token (i.e., the user as you can see in the image of the PR)